### ISSUE TITLE
training tips

### ISSUE THREAD

**Issue by skywo1f** [deepglugs/dalle#5]
Ok so I believe I've "successfully" completed the training process. Unfortunately my generated images are messy canvasses of color. Do you have any tips on how to improve my training? Right now I am using the 70k image stack scraped from wikipedia and a curated vocab of the 500 most common words. I am training vqvae for the default 2 epochs and dalle for the default 2 epochs. Should I train for more epochs? should I trim my data set? do I need more data? any other tips?

**Comment by deepglugs** [comment]
codebook dims and dalle depth are important.  I've trained my danbooru
figures dataset for 5-6 epochs with 20 depth and am bart starting to see
eyes in the results.  2 epochs for dalle is not enough, imo.

**Comment by skywo1f** [comment]
it looks like the default depth is 48 (which ive been using):
    parser.add_argument('--depth', default=48, type=int)

but yeah, I will probably have to trim my dataset. Otherwise 6 epochs will take about a week haha.

**Comment by deepglugs** [comment]
Also make sure the VAE samples look good.  Train until they do.  If they
don't, you'll never get good results from dalle.

**Comment by skywo1f** [comment]
ok so I 
1. narrowed my list down to 76 vocab words
2. I used 26k images for those words
3. trained vae for 10 epochs down to a loss of 0.000563
4. the vae samples looked ok? just looked like regular images
5. then I used that to train dalle for 10 epochs. This took ~5 days. 
6. got it down to a loss of 0.108623

Still, when I ask it for a simple image (blue, flower) it gives me some colorless abstract art . I get similar nonsense when I ask it to generate other samples. 
![my_image](https://user-images.githubusercontent.com/4000822/111150809-f0510f00-8564-11eb-8738-bad3c1285402.png)

Am I
1. missing some big step
2. needing a better training set
3. just bad at training in general

**Comment by skywo1f** [comment]
this is a sample from the last epoch of vae:
![vae_10_1600](https://user-images.githubusercontent.com/4000822/111151255-97ce4180-8565-11eb-99c1-6123f6a8badd.png)

here is a sample from the last epoch of dalle:
![dalle_10_1600](https://user-images.githubusercontent.com/4000822/111151314-a9174e00-8565-11eb-8cbd-0de8095e4c3e.png)

**Comment by deepglugs** [comment]
The samples look good. What is your codebook size?  Looks like the paper
used 1024 size.  26k images isn't a lot compared to the what the paper
trained on.  Try other vocab combinations?

**Comment by skywo1f** [comment]
My codebook size is the default 1024. I had 466 samples of flowers and 437 samples of blue. It didnt even come close to drawing either. I am not sure how changing the vocab will change my results.

**Comment by deepglugs** [comment]
I mean trying "red flower" or "flower blue", see if that changes things.
Also, did you use a clip model?  That will help guide he results.
